request.go

```diff
@@ -11,13 +11,15 @@ type (
 	// Request is an HTTP request to be sent to an OCI registry.
 	Request struct {
 		*resty.Request
+		retryCallback func(*Request)
 	}
 
 	requestConfig struct {
-		Name      string
-		Reference string
-		Digest    string
-		SessionID string
+		Name          string
+		Reference     string
+		Digest        string
+		SessionID     string
+		RetryCallback func(*Request)
```

Comment:
Can we make this return an error, in the case that some callbacks might throw one?

Reply:
Good idea - done. Since the retry callback type is now a little less easy to read, I've given it a name as well.